editor.planx.uk/src/pages/FlowEditor/lib/analyticsProvider.tsx

```diff
@@ -140,43 +144,59 @@ export const AnalyticsProvider: React.FC<{ children: React.ReactNode }> = ({
         trackBackwardsNavigationByNodeId,
         node,
         trackInputErrors,
+        track,
       }}
     >
       {children}
     </Provider>
   );
 
   async function track(
-    direction: AnalyticsLogDirection,
-    analyticsId: number,
     nodeId: string,
+    direction?: AnalyticsLogDirection,
+    analyticsSessionId?: number,
   ) {
     const nodeToTrack = flow[nodeId];
+    const logDirection = direction || detemineLogDirection();
+    const analyticsSession = analyticsSessionId || analyticsId;
 
-    const metadata = getNodeMetadata(nodeToTrack);
+    if (!nodeToTrack || !logDirection || !analyticsSession) {
+      return;
+    }
+
+    const metadata: NodeMetadata = getNodeMetadata(nodeToTrack);
     const nodeType = nodeToTrack?.type ? TYPES[nodeToTrack.type] : null;
     const nodeTitle = extractNodeTitle(nodeToTrack);
+    const isAutoAnswered = breadcrumbs[nodeId]?.auto || false;
```

Comment:
nit: if you're defaulting to `false` when undefined here, it seems like we should perhaps be able to tighten this type then?

can current type definition `isAutoAnswered?: boolean | null;` actually be `isAutoAnswered: boolean;`  based on implementation here?

Reply:
I've updated the logic around this here: c46b9aada1c6272f7a2a6ab39820241805680812

Although, I've left `isAutoAnswered?` as there's somewhere else `NodeMetadata` is used where I don't think it makes sense to look up if the question is auto answered.